dbms/src/Storages/KVStore/Read/LearnerRead.cpp

```diff
@@ -12,523 +12,52 @@
-            auto res = RegionTable::resolveLocksAndWriteRegion(
-                tmt,
-                physical_table_id,
-                region,
-                mvcc_query_info->read_tso,
-                region_to_query.bypass_lock_ts,
-                region_to_query.version,
-                region_to_query.conf_version,
-                log);
-
-            std::visit(
-                variant_op::overloaded{
-                    [&](LockInfoPtr & lock) { unavailable_regions.addRegionLock(region->id(), std::move(lock)); },
-                    [&](RegionException::RegionReadStatus & status) {
-                        if (status != RegionException::RegionReadStatus::OK)
-                        {
-                            LOG_WARNING(
-                                log,
-                                "Check memory cache, region_id={} version={} handle_range={} status={}",
-                                region_to_query.region_id,
-                                region_to_query.version,
-                                RecordKVFormat::DecodedTiKVKeyRangeToDebugString(region_to_query.range_in_table),
-                                magic_enum::enum_name(status));
-                            unavailable_regions.add(region->id(), status);
-                        }
-                    },
-                },
-                res);
-        } // wait index for next region
-        GET_METRIC(tiflash_syncing_data_freshness).Observe(batch_wait_data_watch.elapsedSeconds()); // For DBaaS SLI
-        stats.wait_index_elapsed_ms = watch.elapsedMilliseconds();
-        LOG_DEBUG(
-            log,
-            "Finish wait index | resolve locks, wait_cost={}ms n_regions={} n_unavailable={}",
-            stats.wait_index_elapsed_ms,
-            batch_read_index_req.size(),
-            unavailable_regions.size());
-    }
-
-    // Will be handled in `CoprocessorHandler::execute`.
-    unavailable_regions.tryThrowRegionException(
-        for_batch_cop,
-        context.getDAGContext() ? context.getDAGContext()->is_disaggregated_task : false);
-
-    const auto end_time = Clock::now();
-    const auto time_elapsed_ms = std::chrono::duration_cast<std::chrono::milliseconds>(end_time - start_time).count();
-    // Use info level if read wait index run slow
-    const auto log_lvl = time_elapsed_ms > 1000 ? Poco::Message::PRIO_INFORMATION : Poco::Message::PRIO_DEBUG;
-    LOG_IMPL(
-        log,
-        log_lvl,
-        "[Learner Read] batch read index | wait index"
-        " total_cost={} read_cost={} wait_cost={} n_regions={} n_stale_read={} n_unavailable={}",
-        time_elapsed_ms,
-        stats.read_index_elapsed_ms,
-        stats.wait_index_elapsed_ms,
-        num_regions,
-        stats.num_stale_read,
-        unavailable_regions.size());
+    auto & tmt = context.getTMTContext();
+    LearnerReadWorker worker(mvcc_query_info, tmt, for_batch_cop, is_wn_disagg_read, log);
```

Comment:
I found `doLearnerRead` is also called in `DAGStorageInterpreter`, however, I found the `mvcc_query_info` argument is not handled like `InterpreterSelectQuery`. Why?

Reply:
The `InterpreterSelectQuery` only work for executing `select xxx from table_xx` coming from the TCP client during mock test cases